x-pack/plugins/spaces/server/saved_objects/spaces_saved_objects_client.ts

```diff
@@ -420,4 +422,33 @@ export class SpacesSavedObjectsClient implements SavedObjectsClientContract {
       namespace: spaceIdToNamespace(this.spaceId),
     });
   }
+
+  /**
+   * Returns a generator to help page through large sets of saved objects.
+   *
+   * The generator wraps calls to `SavedObjects.find` and iterates over
+   * multiple pages of results using `_pit` and `search_after`. This will
+   * open a new Point In Time (PIT), and continue paging until a set of
+   * results is received that's smaller than the designated `perPage`.
+   *
+   * @param {object} findOptions - {@link SavedObjectsCreatePointInTimeFinderOptions}
+   * @param {object} [dependencies] - {@link SavedObjectsCreatePointInTimeFinderDependencies}
+   */
+  createPointInTimeFinder(
+    findOptions: SavedObjectsCreatePointInTimeFinderOptions,
+    dependencies?: SavedObjectsCreatePointInTimeFinderDependencies
+  ) {
+    throwErrorIfNamespaceSpecified(findOptions);
```

Comment:
It feels awkward to have the `dependencies` exposed on the public contract to let wrappers override them with their own APIs, OTOH given the approach we decided to take, I don't really know how this could be avoided without dissociating the public SOC API from the one that wrappers need to implement, which seems overcomplicated...

Reply:
This stood out to me too, and yeah, seems like that's the only way we could resolve it. FWIW the SOC already has this issue with the `namespace` option for every SOR/SOC method. We really don't want/need to expose that to other consumers but it's there.